src/Core/Indexes/Octree.cs

```diff
@@ -0,0 +1,104 @@
+﻿namespace AEGIS.Indexes
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    /// <summary>
+    /// Represents a 3D Octree, which contains a collection of <see cref="IBasicGeometry" /> instances.
+    /// </summary>
+    public class Octree : QuadTree
+    {
+        /// <summary>
+        /// Represents a node of the Octree.
+        /// </summary>
+        protected class OctreeNode : QuadTreeNode
+        {
+            /// <summary>
+            /// The children nodes of this node. Each internal node must always have 8 children.
+            /// </summary>
+            private readonly List<OctreeNode> children;
```

Comment:
As `OctreeNode` is a descendant of `QuadTreeNode`, all fields are inherited, which means the children list is also inherited. Each `OctreeNode` instance will have a list of 8 octree nodes and 4 quadtree nodes (which is not used), which results to 40 bytes memory wasted for each node. Please remove this. I suggest using the children list of QuadTreeNode (due to the inheritance, it can also hold `OctreeNode` instances). You can always cast instances using the `as` operator if needed.